### PR TITLE
Add license header to top of all code files

### LINKED ISSUE #749: Add license header to top of all code files
See https://opensource.google/docs/releasing/licenses/

We can either use;
```
Copyright (C) 2020  The SymbiFlow Authors.

Use of this source code is governed by a ISC-style
license that can be found in the LICENSE file or at
https://opensource.org/licenses/ISC
```

Or put the full license at the top;
```
Copyright (C) 2020  The SymbiFlow Authors.

Permission to use, copy, modify, and/or distribute this software for any
purpose with or without fee is hereby granted, provided that the above
copyright notice and this permission notice appear in all copies.

THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
```

### PR DESCRIPTION
**Pull request by tgorochowik**
Fixes #749

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,9 @@
+# Copyright (C) 2020  The SymbiFlow Authors.
+#
+# Use of this source code is governed by a ISC-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/ISC
```
Comment (mithro) [inline Makefile:5]: Please add a `SPDX` specifiers as described in https://github.com/SymbiFlow/ideas/issues/47

File: generators/ariane [hunk 2]
```diff
@@ -1,5 +1,11 @@
 #!/usr/bin/env python3
```
Comment (mithro) [inline generators/ariane:2]: After a bunch of umming and ahhing in the #symbiflow IRC channel we decided that this newline in Python files should have a hash to join the hashbang and the copyright statement.
```suggestion
#
```

File: generators/ariane [hunk 3]
```diff
@@ -1,5 +1,11 @@
...
 
+# Copyright (C) 2020  The SymbiFlow Authors.
+#
+# Use of this source code is governed by a ISC-style
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/ISC
```
Comment (mithro) [inline generators/ariane:7]: Please add a `SPDX` specifier here too.

File: generators/ariane [hunk 4]
```diff
@@ -1,5 +1,11 @@
 #!/usr/bin/env python3
```
Comment (mithro) [inline generators/ariane:1]: Please add the coding line to all the Python files.

```suggestion
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
```

File: tools/BaseRunner.py [hunk 5]
```diff
@@ -1,3 +1,9 @@
+# Copyright (C) 2020  The SymbiFlow Authors.
```
Comment (mithro) [inline tools/BaseRunner.py:1]: For non-runnable Python files I think we want the following (which makes sure that editors detect this file as Python 3?) -- Need to confirm this....
```
#!/bin/false python3
# -*- coding: utf-8 -*-
```

**Comment by tgorochowik** [comment]
A copy from the original task to track progress here:
- [ ]  COPYING file renamed to LICENSE file
- [ ]  The LICENSE file should start with - Copyright 20XX The [Project Name] Authors. -- See https://opensource.google/docs/releasing/authors/
- [ ]  Add or update the AUTHORS files to have the text as shown at https://opensource.google/docs/releasing/authors/
- [ ]  Remove the un-needed All rights reserved. text.
- [ ]  Make sure all code files have the license specifier at the top of the file as shown at https://opensource.google/docs/releasing/licenses/
- [ ]  All code files should have SPDX identifiers.
- [ ]  Add CI which checks for all this.
- [ ]  Make sure that all third_party code is under third_party directory.
- [ ]  Make sure that all third_party directories have a correct LICENSE file.
- [ ]  Python files should have a #!/usr/bin/env python3 as the shebang line.
- [ ]  Python files should have a # coding: utf-8 as the second line after the shebang.

**Comment by tgorochowik** [comment]
@mithro thanks for your comments.

The only thing that is missing from the list in https://github.com/SymbiFlow/ideas/issues/47 is the CI to check for all of this. I guess this should be something common for all the repositories. Do we have something like that somewhere already?

**Comment by kgugala** [comment]
> @mithro thanks for your comments.
> 
> The only thing that is missing from the list in [SymbiFlow/ideas#47](https://github.com/SymbiFlow/ideas/issues/47) is the CI to check for all of this. I guess this should be something common for all the repositories. Do we have something like that somewhere already?

@tgorochowik, @acomodi added a script for checking this in this PR https://github.com/SymbiFlow/fpga-tool-perf/pull/64

**Comment by tgorochowik** [comment]
Added the license checker by @acomodi from #773

**Comment by hzeller** [comment]
What is the status of this one ? It requires rebasing, but other than that, is @mithro happy ?

**Comment by tgorochowik** [comment]
Rebased the changes onto current master. All the requirements are met so I think this is ready (assuming the CI succeeds).

**Comment by mithro** [comment]
@tgorochowik -- Can you rebase this and we can merge it?

File: .github/travis/check_license.sh [hunk 6]
```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
```
Comment (mithro) [inline .github/travis/check_license.sh:1]: Can you update this with the latest version from https://github.com/SymbiFlow/python-symbiflow-v2x/blob/master/.github/ci/check_license.sh ?

File: .travis.yml [hunk 7]
```diff
@@ -20,6 +20,10 @@ jobs:
       script:
```
Comment (mithro) [inline .travis.yml:20]: Please move this to GitHub actions.

File: AUTHORS [hunk 8]
```diff
@@ -0,0 +1,15 @@
+# This is the list of sv-tests's significant contributors.
```
Comment (mithro) [inline AUTHORS:1]: Are there more people who need to be added here now?

**Review by mithro — APPROVED** [review]
Can you rebase this and update the scripts from the copies in python-symbiflow-v2x - then please ping me and I'll merge it.

File: .github/travis/check_license.sh (hunk 6 above)
Comment (tgorochowik) [reply .github/travis/check_license.sh]: Done

File: .travis.yml (hunk 7 above)
Comment (tgorochowik) [reply .travis.yml]: Done

File: AUTHORS (hunk 8 above)
Comment (tgorochowik) [reply AUTHORS:1]: It currently lists people with 5+ commits (excluding people working for Antmicro or Google who are not listed directly)

Comment (mithro) [reply AUTHORS:1]: Great!

**Review by mithro — APPROVED** [review]

**Comment by mithro** [comment]
One minor fix;
```
ERROR: third_party/cores/black-parrot does not have the LICENSE file.
ERROR: third_party/tests/axi_vip does not have the LICENSE file.
Error: Process completed with exit code 1.
```

**Pull request merged**